client/volume_inspect.go

```diff
@@ -18,15 +17,12 @@ func (cli *Client) VolumeInspect(ctx context.Context, volumeID string) (types.Vo
 
 // VolumeInspectWithRaw returns the information about a specific volume in the docker host and its raw representation
 func (cli *Client) VolumeInspectWithRaw(ctx context.Context, volumeID string) (types.Volume, []byte, error) {
-	// The empty ID needs to be handled here because with an empty ID the
-	// request url will not contain a trailing / which calls the volume list API
-	// instead of volume inspect
 	if volumeID == "" {
 		return types.Volume{}, nil, objectNotFoundError{object: "volume", id: volumeID}
 	}
 
 	var volume types.Volume
-	resp, err := cli.get(ctx, path.Join("/volumes", volumeID), nil, nil)
+	resp, err := cli.get(ctx, "/volumes/"+volumeID, nil, nil)
```

Comment:
Why this change? I think using `path.Join()` is correct

Reply:
Looks like we use concatenation in all the other endpoints

Reply:
I changed it to be consistent with the other similar methods. Also for simplicity, I could not see a case where it would be necessary to use `path.Join` over the simple concat, but I maybe overlooking something.

Reply:
I think the only case where it would be different is the empty string value. I guess it's fine, we can fix them all at once at some later time.